### PR TITLE
Image-installer: Fix duplication of image prefix

### PR DESCRIPTION
**Pull request by fastiuk**
* SONiC prefix shown twice when working branch
contains "image-" in its name.

Signed-off-by: Yevhen Fastiuk <yfastiuk@nvidia.com>

#### Why I did it
SONiC prefix is shown twice when the working branch contains "image-" in its name.
Screenshot 2022-05-18 at 14 08 12

#### How I did it
Fixed sonic-installer's one bootloader wrapper to show correct version.
Screenshot 2022-05-18 at 14 08 33

#### How to verify it
Build image (the branch is important):
```
cd sonic-buildimage
git checkout -b dev-image-test
make configure PLATFORM=mellanox && make target/sonic-mellanox.bin
```
Run image on the switch and execute:
```
sudo sonic-installer list
```

#### Previous command output (if the output of a command-line utility has changed)
Screenshot 2022-05-18 at 14 08 12

#### New command output (if the output of a command-line utility has changed)
Screenshot 2022-05-18 at 14 08 33

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@fastiuk please note that new infra was added to test sonic-installer on https://github.com/Azure/sonic-utilities/pull/2179. Suggest you will wait for this PR to get merged and then add test to have this part covered as well.

**Comment by liat-grozovik** [comment]
@stepanblyschak please review

**Comment by liat-grozovik** [comment]
@stepanblyschak i am un able to assign you as a reviewer so sending a reminder :-) thanks.
@fastiuk please add UT please so we can signoff.

File: sonic_installer/bootloader/onie.py [hunk 1]
```diff
@@ -21,7 +21,10 @@ def get_current_image(self):
         cmdline = open('/proc/cmdline', 'r')
         current = re.search(r"loop=(\S+)/fs.squashfs", cmdline.read()).group(1)
         cmdline.close()
-        return current.replace(IMAGE_DIR_PREFIX, IMAGE_PREFIX)
+        # Replace only the first occurrence, since we are using branch name as
+        # tagging for version, IMAGE_PREFIX in the name of the branch may be
+        # replaced as well.
+        return current.replace(IMAGE_DIR_PREFIX, IMAGE_PREFIX, 1)
```
Comment (stepanblyschak) [inline sonic_installer/bootloader/onie.py:27]: Please add unit tests

Comment (fastiuk) [reply sonic_installer/bootloader/onie.py:27]: Done

**Review by stepanblyschak — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by yxieca** [comment]
Is this a regression? If yes, which change introduced regression?

What platform(s) are impacted?

**Comment by fastiuk** [comment]
> Is this a regression?

@yxieca I think this is a zero-day issue. At least I investigated so from git history.

**Comment by yxieca** [comment]
> I think this is a zero-day issue. At least I investigated so from git history.

@fastiuk, I think you have some image that we built, like the ones used for factory default image. I don't observe this issue on these images. Nor did I see this issue before. I am not convinced it is a zero-day issue. I am wondering if this change would cause issue on platforms that wasn't subject to this issue before?

**Comment by yxieca** [comment]
Discussed with Qi. This is a very specific corner case.

**Comment by yxieca** [comment]
This is a good change. But I don't think we need to cherry-pick it to a feature branch. The corner case is very specific.

**Pull request merged**